interfaces/builtin/basedeclaration.go

```diff
@@ -324,6 +324,11 @@ slots:
       slot-snap-type:
         - core
     deny-auto-connection: true
+  joystick:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
```

Comment:
I think your instincts for the base decalaration are correct. On the one hand, this is giving precisely what the interface advertises, but it is a device driver which adds attack surface to the kernel. I also suspect that we are going to need more extensive udev and/or /dev/input access in the future will definitely means we won't want to auto-connect. We can always reconsider in the future.